external/kafka/src/test/scala/org/apache/spark/streaming/kafka/KafkaRDDSuite.scala

```diff
@@ -40,43 +41,70 @@ class KafkaRDDSuite extends KafkaStreamSuiteBase with BeforeAndAfter {
     tearDownKafka()
   }
```

Comment:
True. But, ....

1 is a riskier change, that should be done later (not for this release), dont want to introduce flakiness
2. is a greater refactoring, that is definitely not worth doing now. I am not sure though nested testsuites are a good idea or not. Stickign all the Kafka tests in a single testsuite is something that we can consider, there are pros and cons. 

For now at least we should reuse the kafka harness within the same testsuite. I did that for the other Kafka testsuites. The bigger question we can address in separate PRs meant for next release cycle.